samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,241 @@
+/*
+  Copyright (C) 1997-2016 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+/* Simple program:  Create a native window and attach an SDL renderer */
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <hal/video.h>
+#include "stdio.h"
+#include "string.h"
+#include <SDL.h>
+#include <SDL_ttf.h>
+
+int _exit(int x)
+{
+    debugPrint("_exit called\n");
+    debugPrint( " SDL_Error: %s\n", SDL_GetError() );
+    XSleep(5000);
+    XReboot();
+    return 0;
+}
+
+//Screen dimension constants
+const extern int SCREEN_WIDTH;
+const extern int SCREEN_HEIGHT;
+#define NUM_SPRITES 10
+#define MAX_SPEED   5
+
+static SDL_Rect *positions, *velocities;
+
+/* Call this instead of exit(), so we can clean up SDL: atexit() is evil. */
+static void
+quit(int rc)
+{
+    SDL_VideoQuit();
+    _exit(rc);
+}
+
+SDL_Texture *
+LoadSprite(SDL_Renderer *renderer)
```

Comment:
This sounds more like "CreateTextureFromFont" (there's no "SDL_Sprite" involved, and it doesn't "Load" anything from disk except a font - but that load is implied. The task this does, is creation of new content using parameters)

For reusability, it shoud probably be more like: `CreateTextureFromFont(const char* font_path, unsigned int font_size, SDL_Color font_color, const char* text, ...);`